gridcomps/ExtData3G/ExtDataGridComp_private.F90

```diff
@@ -155,22 +174,45 @@ subroutine set_weights(state, rc)
       type(ESMF_StateItem_Flag), allocatable :: itemTypeList(:)
       integer itemCount,i
       type(ESMF_FieldBundle) :: bundle
-      type(ESMF_Info) :: infoh
-      real :: weights(3)
-      weights = [0.0,0.5,0.5]
+      logical :: is_active
 
       call ESMF_StateGet(state, itemCount=itemCount, _RC)
       allocate(itemNameList(itemCount), _STAT)
       allocate(itemTypeList(itemCount), _STAT)
       call ESMF_StateGet(state, itemTypeList=itemTypeList, itemNameList=itemNameList, _RC)
       do i=1,itemCount
          _ASSERT(itemTypeList(i) == ESMF_STATEITEM_FIELDBUNDLE, 'all items in extdata exprot should be fieldbundles')
-         call ESMF_StateGet(state, itemNameList(i), bundle, _RC)
-         call MAPL_FieldBundleSet(bundle, interpolation_weights=weights, _RC)
+         call ESMF_StateGet(state, trim(itemNameList(i)), bundle, _RC)
+         call MAPL_FieldBundleGet(bundle, is_active=is_active, _RC)
+         if (is_active) call active_list%push_back(trim(itemNameList(i)))
       enddo 
 
       _RETURN(_SUCCESS)
 
-   end subroutine set_weights
+   end function get_active_items
+
+   subroutine report_active_items(exports, lgr)
+      type(PrimaryExportVector), intent(in) :: exports
+      class(logger), pointer :: lgr
+
+      type(PrimaryExportVectorIterator) :: iter
+      type(PrimaryExport), pointer :: export
+      character(len=:), allocatable :: export_name
+      integer :: i
+
+      call lgr%info('*******************************************************')
+      call lgr%info('** Variables to be provided by the ExtData Component **')
```

Comment:
May not want `ExtData` in the comment here.   Most configurations of the logger are going to prepend gridcomp name at beginning of line.